test/tests/blob.js

```diff
@@ -1,16 +1,80 @@
 var assert = require("assert");
 var path = require("path");
 var local = path.join.bind(path, __dirname);
+var promisify = require("promisify-node");
+var fse = promisify("fs-extra");
+var exec = require("../../utils/execPromise");
 
 describe("Blob", function() {
   var NodeGit = require("../../");
 
   var Oid = NodeGit.Oid;
   var Repository = NodeGit.Repository;
+  var Blob = NodeGit.Blob;
   var FileMode = NodeGit.TreeEntry.FILEMODE;
 
   var reposPath = local("../repos/workdir");
   var oid = "111dd657329797f6165f52f5085f61ac976dcf04";
+  var previousCommitOid = "";
+
+  function commitFile(repo, fileName, fileContent, commitMessage) {
+    var index;
+    var treeOid;
+    var parent;
+
+    return fse.writeFile(path.join(repo.workdir(), fileName), fileContent)
+    .then(function() {
+      return repo.refreshIndex();
+    })
+    .then(function(indexResult) {
+      index = indexResult;
+    })
+    .then(function() {
+      return index.addByPath(fileName);
+    })
+    .then(function() {
+      return index.write();
+    })
+    .then(function() {
+      return index.writeTree();
+    })
+    .then(function(oidResult) {
+      treeOid = oidResult;
+      return NodeGit.Reference.nameToId(repo, "HEAD");
+    })
+    .then(function(head) {
+      return repo.getCommit(head);
+    })
+    .then(function(parentResult) {
+      parent = parentResult;
+      return Promise.all([
+        NodeGit.Signature.create("Foo Bar", "foo@bar.com", 123456789, 60),
+        NodeGit.Signature.create("Foo A Bar", "foo@bar.com", 987654321, 90)
+      ]);
+    })
+    .then(function(signatures) {
+      var author = signatures[0];
+      var committer = signatures[1];
+
+      return repo.createCommit(
+        "HEAD",
+        author,
+        committer,
+        commitMessage,
+        treeOid,
+        [parent]);
```

Comment:
I noticed you have a preference for closing on the same line, can we update the style to open/close on new lines if it's necessary to add new lines?